### PR TITLE
Add support for aten.min.dim

### PR DESCRIPTION
**Pull request by cccclai**
Differential Revision: D78526966

### CODE REVIEW

**Comment by cccclai** [comment]
@haowhsu-quic @shewu-quic @DannyYuyang-quic @winskuo-quic I ran into some issues to lower an internal model to QNN and this is one of the missing ops. I think the output are two nodes, while the regular min output is one node. Do you think we should re-use the `op_min` or have a different file to handle this case?

**Comment by cccclai** [comment]
Actually, seems like the min operator in HTP doesn't support returning index, we may need to decompose `min.dim`, is it correct?

**Comment by haowhsu-quic** [comment]
> Actually, seems like the min operator in HTP doesn't support returning index, we may need to decompose `min.dim`, is it correct?

Yes, looks like the most similar one is ReduceMin. But it cannot return index.
Maybe we can compose it with ReduceMin + ArgMin.

**Comment by DannyYuyang-quic** [comment]
> Actually, seems like the min operator in HTP doesn't support returning index, we may need to decompose `min.dim`, is it correct?

Thanks for catching that, I’ll handle the min.dim op support

**Comment by cccclai** [comment]
> > Actually, seems like the min operator in HTP doesn't support returning index, we may need to decompose `min.dim`, is it correct?
> 
> Thanks for catching that, I’ll handle the min.dim op support

Thanks! That will be super helpful. I find that we need to have 2 more ops...can we have the following ops supported?
```
                "aten.min.dim",
                "aten.logical_and.default",
                "aten.max.dim",
```

**Comment by DannyYuyang-quic** [comment]
> > > Actually, seems like the min operator in HTP doesn't support returning index, we may need to decompose `min.dim`, is it correct?
> > 
> > 
> > Thanks for catching that, I’ll handle the min.dim op support
> 
> Thanks! That will be super helpful. I find that we need to have 2 more ops...can we have the following ops supported?
> 
> ```
>                 "aten.min.dim",
>                 "aten.logical_and.default",
>                 "aten.max.dim",
> ```

Yes sure, could you please also share the source(testcase) for `aten.logical_and.default`~?

**Comment by cccclai** [comment]
Something like this will be helpful
```
class LogicalAnd(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, x, y):
        return torch.logical_and(x, y)

a = torch.tensor([0, 1, 10, 0], dtype=torch.int8)
b = torch.tensor([4, 0, 1, 0], dtype=torch.int8)
model = LogicalAnd()
example_inputs = (a, b)
model(*example_inputs)
```

**Comment by DannyYuyang-quic** [comment]
> Something like this will be helpful
> 
> ```
> class LogicalAnd(torch.nn.Module):
>     def __init__(self):
>         super().__init__()
> 
>     def forward(self, x, y):
>         return torch.logical_and(x, y)
> 
> a = torch.tensor([0, 1, 10, 0], dtype=torch.int8)
> b = torch.tensor([4, 0, 1, 0], dtype=torch.int8)
> model = LogicalAnd()
> example_inputs = (a, b)
> model(*example_inputs)
> ```

`aten.min.dim` and `aten.max.dim` are almost ready, I’ll push the PR once internal CI passes.

Regarding `logical_and`, it seems that int8 inputs are not supported,  the `ElementWiseAnd` only accepts UFIXED_POINT_8, SFIXED_POINT_8, or BOOL_8. Is int8 part of your internal model in FP mode?

**Comment by cccclai** [comment]
merged with a different PR
